src/fabric/api/fixed-asset-base/go/contracts/FixedAssetContract.go

```diff
@@ -176,6 +178,52 @@ func (contract *FixedAssetContract) PaginatedRangeQuery(ctx utils.Context, start
 	}, nil
 }
 
+//Do y read and x write - directly returns the string
+func (contract *FixedAssetContract) ReadWriteAssets(ctx utils.Context, readUuids []string, writeUiids []string, letter string) (*[]*assets.FixedAsset, error) {
+	fmt.Println("Entering compositeTx()")
+
+	retArr := make([]*assets.FixedAsset, 0)
+
+	for _, uuid := range readUuids {
+		bytes, err := ctx.GetStub().GetState(uuid)
+
+		if err != nil {
+			fmt.Println("Error performing GetState: " + err.Error())
+			return nil, err
+		}
+
+		fixedAsset := new(assets.FixedAsset)
+
+		err = json.Unmarshal(bytes, fixedAsset)
+
+		if err != nil {
+			fmt.Println("Error performing json.Unmarshal: " + err.Error())
+			fmt.Println("Error performing json.Unmarshal on bytes: " + string(bytes[:]))
+			return nil, err
+		}
+
+		retArr = append(retArr, fixedAsset)
+	}
+
+	assetTemplate := retArr[0]
+	byteSize := retArr[0].Bytesize
+	for _, uuid := range writeUiids {
+		assetTemplate.UUID = uuid
+		paddingSize := byteSize - int(unsafe.Sizeof(assetTemplate))
+		assetTemplate.Content = strings.Repeat(letter, paddingSize)
+		bytes, _ := json.Marshal(assetTemplate)
+		err := ctx.GetStub().PutState(uuid, bytes)
+		if err != nil {
+			fmt.Println("Error performing PutState: " + err.Error())
+			return nil, err
+		}
+	}
+
+	fmt.Println(`Exiting compositeTx(), returning result set of size: ${retArr.length}`)
+
+	return &retArr, nil
```

Comment:
I don't think we need to worry about the returning of data or maybe make it optional with the default being to not return